src/main/java/com/cloudbees/jenkins/support/impl/RunDirectoryComponent.java

```diff
@@ -97,6 +91,26 @@ public boolean understandsSymlink() {
         }
     }
 
+    private static String getBuildRelativePath(@NonNull Run item, @NonNull File itemRootDir) {
+
+        // See Jenkins#isDefaultBuildDir that has restriction and cannot be used outside Jenkins core
+        File buildsRootDir = "${ITEM_ROOTDIR}/builds".equals(Jenkins.get().getRawBuildsDir())
+                ? new File(Jenkins.get().getRootDir(), "jobs")
+                // Not using the default buildsDir, find the custom root dir by keeping what's before the compulsory
+                // ${ITEM_FULL_NAME} placeholder
+                : new File(Objects.requireNonNull(Util.replaceMacro(
+                        Jenkins.get().getRawBuildsDir().replaceFirst("\\$\\{ITEM_FULL_NAME.*", ""),
+                        Map.of("JENKINS_HOME", Jenkins.get().getRootDir().getPath()))));
+
+        return Functions.isWindows()
+                ? buildsRootDir
+                        .toPath()
+                        .relativize(itemRootDir.toPath())
+                        .toString()
+                        .replace('\\', '/')
+                : buildsRootDir.toPath().relativize(itemRootDir.toPath()).toString();
```

Comment:
or just
```suggestion
        return buildsRootDir
                        .toPath()
                        .relativize(itemRootDir.toPath())
                        .toString()
                        .replace(File.separatorChar, '/');
```
since `String.replace(char, char)` is a fast no-op if the args are equal.

Reply:
Yeah, makes sense.